cinema/tests/test_movie_view_set.py

```diff
@@ -0,0 +1,116 @@
+
+    return Movie.objects.create(**defaults)
+
+
+def sample_genre(**params):
+    defaults = {
+        "name": "Drama",
+    }
+    defaults.update(params)
+
+    return Genre.objects.create(**defaults)
+
+
+def sample_actor(**params):
+    defaults = {"first_name": "George", "last_name": "Clooney"}
+    defaults.update(params)
+
+    return Actor.objects.create(**defaults)
+
+
+def sample_movie_session(**params):
+    cinema_hall = CinemaHall.objects.create(name="Blue", rows=20, seats_in_row=20)
+
+    defaults = {
+        "show_time": "2022-06-02 14:00:00",
+        "movie": None,
+        "cinema_hall": cinema_hall,
+    }
+    defaults.update(params)
+
+
+class MovieViewSetTests(APITestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test@test.test", password="testpassword"
+        )
+        self.client.force_authenticate(self.user)
+        self.genre = sample_genre()
+        self.actor = sample_actor()
+        self.movie = sample_movie()
+        self.movie.genres.add(self.genre)
+        self.movie.actors.add(self.actor)
+        self.movie.save()
+
+    def test_list_movies(self):
+        url = (MOVIE_URL)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_retrieve_movie(self):
+        url = reverse("cinema:movie-detail", args=[self.movie.id])
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_filter_movies_by_title(self):
+        url = (MOVIE_URL) + f"?title={self.movie.title}"
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], self.movie.title)
+
+    def test_filter_movies_by_genre(self):
+        url = (MOVIE_URL) + f"?genres={self.genre.id}"
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], self.movie.title)
+
+    def test_filter_movies_by_actor(self):
+        url = (MOVIE_URL + f"?actors={self.actor.id}")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]["title"], self.movie.title)
+
+    def test_get_serializer_class(self):
+        view = MovieViewSet()
+        view.action = "list"
+        self.assertEqual(view.get_serializer_class(), MovieListSerializer)
+
+        view.action = "retrieve"
+        self.assertEqual(view.get_serializer_class(), MovieDetailSerializer)
+
+        view.action = "upload_image"
+        self.assertEqual(view.get_serializer_class(), MovieImageSerializer)
+
+        view.action = ""
```

Comment:
It's a good practice to test the default case in your switch-case structure. In this case, you should add a test to check what happens when `view.action` is neither 'list', 'retrieve', nor 'upload_image'.